python-threatexchange/threatexchange/fetcher/fetch_api.py

```diff
@@ -111,22 +112,28 @@ def get_own_owner_id(self, collab: TCollabConfig) -> int:
         """
         return -1
 
-    def fetch_once(
+    @abstractmethod
+    def fetch_iter(
         self,
-        supported_signal_types: t.List[t.Type[SignalType]],
+        supported_signal_types: t.Sequence[t.Type[SignalType]],
         collab: TCollabConfig,
         # None if fetching for the first time,
         # otherwise the previous FetchDelta returned
         checkpoint: t.Optional[state.TFetchCheckpoint],
-    ) -> TFetchDelta:
+    ) -> t.Iterator[TFetchDelta]:
```

Comment:
Not sure we are on the same page..

```python
x: t.Iterator[TFetchDelta] = fetch_iter()
``` 

`len(x)` should error out. There is no way to know without greedy fetching how many more delta objects can be obtained from the API. And I believe we want to avoid greedy fetching.

However, for each TFetchDelta, there len(delta.updates) should return an integer, and that is fine.